soc/openhwgroup/cva6/cva6.h

```diff
@@ -0,0 +1,16 @@
+/*
+ * OpenHwGroup cva6 test harness success/failure callback
+ *
+ * Copyright(c) 2024, CISPA Helmholtz Center for Information Security
+ * SPDX - License - Identifier : Apache-2.0
+ */
+#pragma once
+
+#ifndef CVA6_H
+#define CVA6_H
+
+#include <stdint.h>
+
+void z_cva6_finish_test(const int32_t status);
```

Comment:
One of the "board" configurations from the other PRs describes a verification harness in which the CVA6 CPU is run in verilator (a HDL simulator). This is used to run test programs in the simulator and check whether the CPU behaves as expected. To this end, the simulator provides an interface that allows completing the test case successfully or with an error code. This is exposed via this SoC-specific function.
The samples I added in https://github.com/zephyrproject-rtos/zephyr/pull/81017 show how this can be used in software.

Reply:
I have added a comment that explains what the function does.

Reply:
Should I remove this functionality from this PR and re-introduce it with the board that actually uses this?

Reply:
@tejlmand